aten/src/ATen/native/cpu/SoftMaxKernel.cpp

```diff
@@ -33,15 +41,19 @@ inline void _vec_log_softmax_lastdim(
     int64_t outer_size,
     int64_t dim_size) {
   using Vec = vec::Vectorized<vec::vec_scalar_t<scalar_t>>;
-  static constexpr int64_t CHUNK_SIZE = (128 / sizeof(scalar_t)) * Vec::size();
-  int64_t grain_size = internal::GRAIN_SIZE / (16 * dim_size * CHUNK_SIZE);
-  if (grain_size < CHUNK_SIZE)
-    grain_size = CHUNK_SIZE;
-
+  // Since number of threads can change at runtime, this variable is not static.
+  int64_t CHUNK_SIZE = get_num_threads();
```

Comment:
May I know why you set the `CHUNK_SIZE` as the number of threads? I guess the `CHUNK_SIZE` was set to improve the cache locality with a proper size while it should not set as a fixed size since the unit of working size is `CHUNK_SIZE * dim_size`, not `CHUNK_SIZE` alone. The remaining logic of the original code that computes the `grain_size` and set it to `CHUNK_SIZE` if it is too small sounds fine. Would it work if we make sure `1 <= CHUNK_SIZE < SOME_FIXED_NUM_BYTES / (dim_size * sizeof(scalar_t))`? Here `SOME_FIXED_NUM_BYTES` stands for a heuristically cache-friendly size.

Reply:
Thanks for your inputs, @jgong5!

 > May I know why you set the CHUNK_SIZE as the number of threads?

 Just to demonstrate way higher speedup. I'll analyze the kernel on Friday & use also use `dim_size` in the computation of `CHUNK_SIZE`. I had marked this PR as draft because I was sure we could optimize it further by considering `dim_size` into the computation of `CHUNK_SIZE`.
 
>  Would it work if we make sure 1 <= CHUNK_SIZE < SOME_FIXED_NUM_BYTES / (dim_size * sizeof(scalar_t))

Thank you for your advice! Yes, something like this should work, and I noticed that some kernels in the same file are using this approach. However, I had planned to write some linear equations in `dim_size` & the total number of threads to arrive at `CHUNK_SIZE`.